milli/src/external_documents_ids.rs

```diff
@@ -49,30 +44,24 @@ impl<'a> ExternalDocumentsIds<'a> {
 
     pub fn delete_ids<A: AsRef<[u8]>>(&mut self, other: fst::Set<A>) -> fst::Result<()> {
         let other = fst::Map::from(other.into_fst());
-        let union_op = self.soft.op().add(&other).r#union();
+        let difference_op = self.docids.op().add(&other).r#difference();
 
-        let mut iter = union_op.into_stream();
-        let mut new_soft_builder = fst::MapBuilder::memory();
+        let mut iter = difference_op.into_stream();
+        let mut new_docids_builder = fst::MapBuilder::memory();
         while let Some((external_id, docids)) = iter.next() {
-            if docids.iter().any(|v| v.index == 1) {
-                // If the `other` set returns a value here it means
-                // that it must be marked as deleted.
-                new_soft_builder.insert(external_id, DELETED_ID)?;
-            } else {
-                let value = docids.iter().find(|v| v.index == 0).unwrap().value;
-                new_soft_builder.insert(external_id, value)?;
-            }
+            let value = docids.iter().find(|v| v.index == 0).unwrap().value;
```

Comment:
Could you add a comment above this `iter.find` call that explains what you are trying to retrieve, please?